nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocumentBreadcrumbJsonEnricher.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (C) Copyright 2022 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.compound.documents;
+
+import static org.nuxeo.compound.documents.CompoundDocumentUtils.COMPOUND_DOCTYPE;
+import static org.nuxeo.compound.documents.CompoundDocumentUtils.COMPOUND_FOLDER_DOCTYPE;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.io.marshallers.json.AbstractJsonWriterTest;
+import org.nuxeo.ecm.core.io.marshallers.json.JsonAssert;
+import org.nuxeo.ecm.core.io.marshallers.json.document.DocumentModelJsonWriter;
+import org.nuxeo.ecm.core.io.registry.context.RenderingContext.CtxBuilder;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+/** @since 2021.0 */
+@RunWith(FeaturesRunner.class)
+@Features(CompoundDocumentsFeature.class)
+@RepositoryConfig(cleanup = Granularity.METHOD)
+public class TestCompoundDocumentBreadcrumbJsonEnricher
+        extends AbstractJsonWriterTest<DocumentModelJsonWriter, DocumentModel> {
+
+    @Inject
+    protected CoreSession session;
+
+    protected DocumentModel document;
+
+    public TestCompoundDocumentBreadcrumbJsonEnricher() {
+        super(DocumentModelJsonWriter.class, DocumentModel.class);
+    }
+
+    @Before
+    public void setup() {
+        document = session.createDocumentModel("/", "level1", COMPOUND_DOCTYPE);
+        document = session.createDocument(document);
+        document = session.createDocumentModel("/level1", "level2", COMPOUND_FOLDER_DOCTYPE);
+        document = session.createDocument(document);
+        document = session.createDocumentModel("/level1/level2", "level3", COMPOUND_DOCTYPE);
+        document = session.createDocument(document);
+        document = session.createDocumentModel("/level1/level2/level3", "level4", "File");
+        document = session.createDocument(document);
+    }
```

Comment:
I would be in favor to not do a setup step as there's only one test, and put the document creation directly in test with some naming change like `level1Doc`, `level2Doc`, etc...